api/models/issue.js

```diff
@@ -0,0 +1,42 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const commentSchema = new Schema({
+    content: { type: String, required: true }
+});
+
+const fileSchema = new Schema({
+    name: { type: String },
+    path: { type: String }
+});
+
+const issueSchema = new Schema(
+    {
+        _id: mongoose.Schema.Types.ObjectId,
+        description: { type: String, required: true },
+        status: { type: Boolean, default: false },
+        issueFiles: [fileSchema],
+        comments: [commentSchema]
+    },
+    {
+        toJSON: {
+            versionKey: false,
+            transform: (doc, ret) => {
+                return {
+                    id: ret._id,
+                    status: ret.status,
+                    description: ret.description,
+                    resourceUrl: process.env.SERVER_URL + 'issues/' + ret._id,
+                    issueFiles: ret.issueFiles,
+                    comments: (() => {
```

Comment:
Nicely written BUT... This create ambiguity in your code. Why? Because if the comments are present attribute `comments` will point to array, something like this:
```
comments: [a, b, c]
```
But if there are no comments, attribute `comments` will point to a string, like this:
```
comments: 'No comments for this issue`
```

This can be dangerous because imagine a code receiving this object and doing something like this:
```
if (issue.comments.length > 0) {
   console.log('this issue has comments');
}
```
if the comments is string 'No comments...' upper code will print wrong information.

Does that make sense?